lms/djangoapps/commerce/utils.py

```diff
@@ -109,3 +117,247 @@ def upgrade_url(self, user, course_key):
+                return False
+    return True
+
+
+def _send_refund_notification(course_product, refund_ids):
+    """ Notify the support team of the refund request. """
+
+    tags = ['auto_refund']
+
+    if theming_helpers.is_request_in_themed_site():
+        # this is not presently supported with the external service.
+        raise NotImplementedError("Unable to send refund processing emails to support teams.")
+
+    student = course_product.user
+    subject = _("[Refund] User-Requested Refund")
+    body = _generate_refund_notification_body(student, refund_ids)
+    requester_name = student.profile.name or student.username
+    create_zendesk_ticket(requester_name, student.email, subject, body, tags)
+
+
+def _generate_refund_notification_body(student, refund_ids):  # pylint: disable=invalid-name
+    """ Returns a refund notification message body. """
+    msg = _(
+        'A refund request has been initiated for {username} ({email}). '
+        'To process this request, please visit the link(s) below.'
+    ).format(username=student.username, email=student.email)
+
+    ecommerce_url_root = configuration_helpers.get_value(
+        'ECOMMERCE_PUBLIC_URL_ROOT', settings.ECOMMERCE_PUBLIC_URL_ROOT,
+    )
+    refund_urls = [urljoin(ecommerce_url_root, '/dashboard/refunds/{}/'.format(refund_id))
+                   for refund_id in refund_ids]
+
+    # emails contained in this message could contain unicode characters so encode as such
+    return u'{msg}\n\n{urls}'.format(msg=msg, urls='\n'.join(refund_urls))
+
+
+def create_zendesk_ticket(requester_name, requester_email, subject, body, tags=None):
+    """ Create a Zendesk ticket via API. """
+    if not (settings.ZENDESK_URL and settings.ZENDESK_USER and settings.ZENDESK_API_KEY):
+        log.debug('Zendesk is not configured. Cannot create a ticket.')
+        return
+
+    # Copy the tags to avoid modifying the original list.
+    tags = list(tags or [])
+    tags.append('LMS')
+
+    # Remove duplicates
+    tags = list(set(tags))
+
+    data = {
+        'ticket': {
+            'requester': {
+                'name': requester_name,
+                'email': unicode(requester_email)
+            },
+            'subject': subject,
+            'comment': {'body': body},
+            'tags': tags
+        }
+    }
+
+    # Encode the data to create a JSON payload
+    payload = json.dumps(data)
+
+    # Set the request parameters
+    url = urljoin(settings.ZENDESK_URL, '/api/v2/tickets.json')
+    user = '{}/token'.format(settings.ZENDESK_USER)
+    pwd = settings.ZENDESK_API_KEY
+    headers = {'content-type': 'application/json'}
+
+    try:
+        response = requests.post(url, data=payload, auth=(user, pwd), headers=headers)
+
+        # Check for HTTP codes other than 201 (Created)
+        if response.status_code != 201:
+            log.error('Failed to create ticket. Status: [%d], Body: [%s]', response.status_code, response.content)
```

Comment:
We need to somehow communicate to `_process_refund` that the ticket was not created successfully, otherwise it will think all is well and move forward with revoke/unenroll.